src/librustc_codegen_ssa/mir/block.rs

```diff
@@ -228,16 +228,6 @@ impl<'a, 'tcx: 'a, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
         &mut self,
         mut bx: Bx,
     ) {
-        if self.fn_ty.c_variadic {
-            match self.va_list_ref {
-                Some(va_list) => {
-                    bx.va_end(va_list.llval);
```

Comment:
Does this pass the codegen tests? I thought I remembered that `Drop` was not called for the "spoofed" `va_list`.

Reply:
`Drop` now gets called for the `VaListImpl`, since it's an actual structure with a `Drop` implementation. The tests check for this, and pass. I even checked some LLVM IR manually, and it looked OK.

Reply:
:+1: awesome